File: ios/Classes/extension/Extensions.swift
import Foundation
import OSAMCommon

private let catalanLanguageCode = "ca"
private let spanishLanguageCode = "es"
private let englishLanguageCode = "en"

extension Language {
    func toLanguageCode() -> String {
        switch self {
        case .ca:
            return catalanLanguageCode
        case .es:
            return spanishLanguageCode
        case .en:
            return englishLanguageCode
        default:
            return catalanLanguageCode
        }
    }
}

func getLanguageFromString(langugageCode: String) -> Language {
    if langugageCode == catalanLanguageCode {
        return Language.ca
    } else if langugageCode == spanishLanguageCode {
        return Language.es
    } else if langugageCode == englishLanguageCode {
        return Language.en
    } else {
        return Language.ca
    }
}

extension VersionControlResponse {
    func toStringResponse() -> String {
        switch self {
        case .accepted:
            return "ACCEPTED"
        case .dismissed:
            return "DISMISSED"
        case .cancelled:
            return "CANCELLED"
        case .error:
            return "ERROR"
        default:
            return "DISMISSED"
        }
    }
}

extension RatingControlResponse {
    func toStringResponse() -> String {
        switch self {
        case .accepted:
            return "ACCEPTED"
        case .dismissed:
            return "DISMISSED"
        case .cancelled:
            return "CANCELLED"
        case .later:
            return "LATER"
        case .error:
            return "ERROR"
        default:
            return "DISMISSED"
        }
    }
}

File: android/src/main/kotlin/cat/bcn/commonmodule/flutter/common_module_flutter/extension/Extensions.kt
package cat.bcn.commonmodule.flutter.common_module_flutter.extension

import cat.bcn.commonmodule.ui.versioncontrol.Language
import cat.bcn.commonmodule.ui.versioncontrol.RatingControlResponse
import cat.bcn.commonmodule.ui.versioncontrol.VersionControlResponse

private const val catalanLanguageCode = "ca"
private const val spanishLanguageCode = "es"
private const val englishLanguageCode = "en"

fun Language.toLanguageCode(): String = when (this) {
    Language.CA -> catalanLanguageCode
    Language.ES -> spanishLanguageCode
    Language.EN -> englishLanguageCode
}

fun getLanguageFromString(languageCode: String): Language = when (languageCode) {
    catalanLanguageCode -> Language.CA
    spanishLanguageCode -> Language.ES
    englishLanguageCode -> Language.EN
    else -> Language.CA
}

fun VersionControlResponse.toStringResponse(): String = when (this) {
    VersionControlResponse.ACCEPTED -> "ACCEPTED"
    VersionControlResponse.DISMISSED -> "DISMISSED"
    VersionControlResponse.CANCELLED -> "CANCELLED"
    VersionControlResponse.ERROR -> "ERROR"
}

fun RatingControlResponse.toStringResponse(): String = when (this) {
    RatingControlResponse.ACCEPTED -> "ACCEPTED"
    RatingControlResponse.DISMISSED -> "DISMISSED"
    RatingControlResponse.CANCELLED -> "CANCELLED"
    RatingControlResponse.LATER -> "LATER"
    RatingControlResponse.ERROR -> "ERROR"
}